sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializerBuilder.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.google.gson.Gson;
+
+/**
+ * Fluent builder class that configures and instantiates instances of {@link GsonJsonSerializer}.
+ */
+public class GsonJsonSerializerBuilder {
+    private Gson gson;
+
+    /**
+     * Constructs a new instance of {@link GsonJsonSerializer} with the configurations set in this builder.
+     *
+     * @return A new instance of {@link GsonJsonSerializer}.
+     */
+    public GsonJsonSerializer build() {
+        return (gson == null)
+            ? new GsonJsonSerializer(new Gson())
+            : new GsonJsonSerializer(gson);
+    }
+
+    /**
+     * Sets the {@link Gson} that will be used during serialization.
+     * <p>
+     * If this is set to {@code null} the default {@link Gson} will be used.
+     *
+     * @param gson {@link Gson} that will be used during serialization.
+     * @return The updated GsonJsonSerializerBuilder class.
+     */
+    public GsonJsonSerializerBuilder gson(Gson gson) {
```

Comment:
Let's find a better name for the setter that sets the implementation in these builders.

Reply:
I'll play with some API names since this will go against the guideline that setters should be named the same as the parameter type.

Reply:
Thoughts on converging both the GSON and Jackson builders to use `serializer` when configuring the `Gson` or `ObjectMapper` respectively.

Reply:
yeah, I thought about `serializer` or `implementation`. Neither are great, but it probably makes sense to just have a standard name for all builders.

Reply:
Updated to use `serializer`.